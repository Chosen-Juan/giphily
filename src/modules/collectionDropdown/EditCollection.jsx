import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { deleteCollection, renameCollection } from '../collection/CollectionActions';
import Gif from '../gifList/Gif';

export class EditCollection extends Component {
  static propTypes = {
    closeCallback: PropTypes.func.isRequired,
    deleteCollectionDispatcher: PropTypes.func.isRequired,
    renameCollectionDispatcher: PropTypes.func.isRequired,
    activeCollectionIndex: PropTypes.number.isRequired,
    collections: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
      items: PropTypes.arrayOf(Gif.WrappedComponent.propTypes.gif)
    }).isRequired)
  };

  constructor(props) {
    super(props);
    const collection = this.props.collections[this.props.activeCollectionIndex];
    this.state = { collection, collectionName: collection.name };
  };

  onInputChange = event => {
    this.setState({ collectionName: event.target.value });
  };

  save = () => {
    this.props.renameCollectionDispatcher(this.state.collection.id, this.state.collectionName);
    this.props.closeCallback();
  };

  delete = () => {
    this.props.deleteCollectionDispatcher(this.props.activeCollectionIndex);
    this.props.closeCallback();
  }

  render() {
    return (
      <div>
        <input
          className='input'
          type='text'
          value={this.state.collectionName}
          onChange={this.onInputChange}
        />
        <div className='buttons'>
          <button className='button is-primary is-outlined' onClick={this.save}>
            Save
          </button>
          <button className='button is-danger is-outlined' onClick={() => this.props.closeCallback()}>
            Cancel
          </button>
          <button className='button is-danger is-outlined' onClick={this.delete}>
            Delete
          </button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = ({ collectionReducer: { activeCollectionIndex, collections }}) => ({
  activeCollectionIndex, collections
});

const mapDispatchToProps = dispatch => ({
  deleteCollectionDispatcher: index => dispatch(deleteCollection(index)),
  renameCollectionDispatcher: (id, name) => dispatch(renameCollection(id, name))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditCollection);
