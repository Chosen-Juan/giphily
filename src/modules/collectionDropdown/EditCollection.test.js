import React from 'react';
import { shallow } from 'enzyme';

import { EditCollection } from './EditCollection';

const spy = jest.fn();
const defaultProps = {
  switchCollectionDispatcher: spy,
  activeCollectionIndex: 0,
  closeCallback: spy,
  deleteCollectionDispatcher: spy,
  renameCollectionDispatcher: spy,
  collections: [{
    id: 'abc-123',
    name: 'Default',
    items: []
  }]
};

describe('EditCollection', () => {
  it('should match snapshot', () => {
    const editCollection = shallow(<EditCollection {...defaultProps} />);
    expect(editCollection).toMatchSnapshot();
  });
});
