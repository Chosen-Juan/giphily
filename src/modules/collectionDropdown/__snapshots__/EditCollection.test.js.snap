// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCollection should match snapshot 1`] = `
<div>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value="Default"
  />
  <div
    className="buttons"
  >
    <button
      className="button is-primary is-outlined"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button is-danger is-outlined"
      onClick={[Function]}
    >
      Cancel
    </button>
    <button
      className="button is-danger is-outlined"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
